### PR TITLE
[WIP] Dartified displaying-data

### PR DESCRIPTION
**Pull request by kwalrath**
Still a couple of PENDING items, but this Dart version could otherwise be close to final, pending technical review.

@tomplusplus I made a few changes to the samples, which you might find interesting. Nothing too hard, though. I just futzed with stuff as it became clear how I was supposed to use the code samples. I think the TS samples have/had some docregions they weren't using, so I deleted some (I might have missed some).

### CODE REVIEW

**Comment by kwalrath** [comment]
@sandfordredlich this is ready for review (with special attention to the PENDING items).

File: public/docs/dart/latest/guide/displaying-data.jade [hunk 1]
```diff
@@ -1,472 +1,265 @@
...
+- var stylePattern = [{ otl: /(platform_directives.*$)/gm }, null, null];
++makeTabs('displaying-data/dart/pubspec.yaml, displaying-data/dart/web/index.html, displaying-data/dart/web/main.dart', ',,final', 'pubspec.yaml, web/index.html, web/main.dart', stylePattern)
+
+:marked
+  All of this code should look familiar from the
+  [QuickStart](../quickstart.html),
+  except for the `platform_directives` entry in `pubspec.yaml`
+  and the imports in `main.dart`.
+
+  In `pubspec.yaml`, the `platform_directives` entry lets us use
```
Comment (matanlurey) [inline public/docs/dart/latest/guide/displaying-data.jade]: Is it possible to link to documentation on the transformer here for people who are interested further? This is definitely new to me :)

File: public/docs/dart/latest/guide/displaying-data.jade [hunk 2]
```diff
@@ -1,472 +1,265 @@
...
+
+  Angular duplicates the `<li>` for each item in the list, setting the `hero` variable
+  to the item (the hero) in the current iteration. Angular uses that variable as the
+  context for the interpolation in the double curly braces.
+
+.l-sub-section
+  :marked
+    We happened to give `NgFor` a list to display.
+    In fact, `NgFor` can repeat items for any [Iterable](https://api.dartlang.org/stable/dart-core/Iterable-class.html) object.
+    **[QUESTION: that used to link to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols. Is Iterable indeed what's supported in Dart?]**
```
Comment (matanlurey) [inline public/docs/dart/latest/guide/displaying-data.jade]: That's what it looks like, anything that implements `dart:core:Iterable` works.

File: public/docs/dart/latest/guide/displaying-data.jade (hunk 1 above)
Comment (kwalrath) [inline public/docs/dart/latest/guide/displaying-data.jade]: The transformer docs I found (https://github.com/angular/angular/wiki/Angular-2-Dart-Transformer) don't have any info on platform_directives, so I'll file bugs to update those docs and to either make these docs point there or add docs here about the transformer.

**Comment by kwalrath** [comment]
About to merge... Thanks very much for the review, @matanlurey!

**Comment by kwalrath** [comment]
Bugs filed: https://github.com/angular/angular/issues/6956, https://github.com/angular/angular.io/issues/825

**Pull request closed without merging**